PowerPointLabs/PowerPointLabs/Models/PowerPointMagnifyingSlide.cs

```diff
@@ -146,9 +145,33 @@ private void AddZoomSlideCroppedPicture(float magnifyRatio = 1.0f)
             cropShape.Select();
             PowerPoint.Selection sel = Globals.ThisAddIn.Application.ActiveWindow.Selection;
             PowerPoint.Shape croppedShape = CropToShape.Crop(zoomSlideCopy, sel, magnifyRatio: magnifyRatio);
-            croppedShape.Cut();
+            try
+            {
+                string tempFilePath = FileDir.GetTemporaryPngFilePath();
+                Utils.GraphicsUtil.ExportShape(croppedShape, tempFilePath);
+                zoomSlideCroppedShapes = _slide.Shapes.AddPicture2(tempFilePath,
+                    Office.MsoTriState.msoFalse,
+                    Office.MsoTriState.msoTrue,
+                    0,
+                    0);
+                croppedShape.Delete();
+                try
+                {
+                    FileDir.DeleteFile(tempFilePath);
+                }
+                catch (Exception)
+                {
+                    // If the file cannot be deleted, we continue without deletion.
+                }
+
+            }
+            catch (Exception)
+            {
+                // Revert to normal copy and pasting if unable to create file.
+                croppedShape.Cut();

```

Comment:
Inconsistent SLAP. The above code in the try block sounds like some special/modified routine to copy/paste a shape. Might it be better if the above code is extracted to a function, to match the level of detail from `croppedShape.Cut()`?